trieste/acquisition/function.py

```diff
@@ -524,3 +527,194 @@ def prepare_acquisition_function(
         :param models: The models over each dataset in ``datasets``.
         :return: A batch acquisition function.
         """
+
+
+class SingleModelBatchAcquisitionBuilder(BatchAcquisitionFunctionBuilder):
+    """
+    Convenience acquisition function builder for an acquisition function (or component of a
+    composite acquisition function) that requires only one model, dataset pair.
+    """
+
+    def using(self, tag: str) -> BatchAcquisitionFunctionBuilder:
```

Comment:
this is a large chunk of repeated code but I don't know how to avoid it

Reply:
i've finally has a chance to sit down and really think this through. Let's chat, cos it's complicated